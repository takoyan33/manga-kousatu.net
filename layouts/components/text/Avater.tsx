/* eslint-disable react/display-name */
import React from "react";
import Avatar from "@mui/material/Avatar";

type Props = {
  photoURL: string;
  displayname: string;
};

//React.memoåŒ–
const Avater: React.VFC<Props> = React.memo(({ photoURL, displayname }) => {
  return (
    <div>
      <Avatar alt="Remy Sharp" src={photoURL} />
      <span className="text-xl">{displayname}</span>
    </div>
  );
});

export default Avater;
