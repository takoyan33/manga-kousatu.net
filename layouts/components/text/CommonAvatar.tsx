/* eslint-disable react/display-name */
import Avatar from '@mui/material/Avatar'
import React from 'react'

type Props = {
  photoURL: string
  displayname: string
}

//React.memoåŒ–
const CommonAvatar = React.memo(({ photoURL, displayname }: Props) => {
  return (
    <>
      <Avatar alt='Remy Sharp' src={photoURL} />
      <span className='text-xl'>{displayname}</span>
    </>
  )
})

export default CommonAvatar
