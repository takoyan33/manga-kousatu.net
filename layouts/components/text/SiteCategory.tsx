import { DOMAttributes, FC, memo, ReactNode } from "react";
import Link from "next/link";
import Button from "@mui/material/Button";

type Props = {
  href: string;
  text: string;
  className: string;
};

// eslint-disable-next-line react/display-name
export const SiteCategory: React.VFC<Props> = memo(
  ({ href, text, className }) => {
    return (
      <span className={className}>
        <Link href={href}>{text}</Link>
      </span>
    );
  }
);

SiteCategory.displayName = "SiteCategory";
