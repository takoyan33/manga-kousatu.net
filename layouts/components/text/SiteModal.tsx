import { toast } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";

export const notify = (text: string) =>
  toast.success(text, {
    position: "top-center",
    autoClose: 5000,
    hideProgressBar: false,
    closeOnClick: true,
    pauseOnHover: true,
    draggable: true,
    progress: undefined,
    theme: "light",
  });

export const signupmissnotify = (text: string) =>
  toast.error(text, {
    position: "top-center",
    autoClose: 5000,
    hideProgressBar: false,
    closeOnClick: true,
    pauseOnHover: true,
    draggable: true,
    progress: undefined,
    theme: "light",
  });
