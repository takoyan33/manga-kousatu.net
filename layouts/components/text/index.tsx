export * from "./Avater";
export * from "./Categori";
export * from "./SiteCategory";
