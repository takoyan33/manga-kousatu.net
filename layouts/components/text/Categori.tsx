/* eslint-disable react/display-name */
import React from "react";
import Link from "next/link";
import { SiteCategory } from "./SiteCategory";

type Props = {
  categori: String;
};

//React.memo化
const Categori: React.VFC<Props> = React.memo(({ categori }) => {
  return (
    <div className="my-1">
      {categori == "ONEPIECE" && (
        <SiteCategory
          className="bg-blue-500 p-1 inline-block text-white text-center hover:bg-blue-700"
          text="ONE PIECE"
          href={`/post/categories/${categori}`}
        />
      )}
      {categori == "呪術廻戦" && (
        <SiteCategory
          className="bg-purple-500 p-1 inline-block text-white text-center hover:bg-purple-700"
          text="呪術廻戦"
          href={`/post/categories/${categori}`}
        />
      )}
      {categori == "東京リベンジャーズ" && (
        <SiteCategory
          className="bg-rose-500 p-1 inline-block text-white text-center hover:bg-yellow-700"
          text="東京リベンジャーズ"
          href={`/post/categories/${categori}`}
        />
      )}
      {categori == "キングダム" && (
        <SiteCategory
          className="bg-yellow-500 p-1 inline-block text-white text-center"
          text="キングダム"
          href={`/post/categories/${categori}`}
        />
      )}
    </div>
  );
});

export default Categori;
