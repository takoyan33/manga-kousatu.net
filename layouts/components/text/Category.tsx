/* eslint-disable react/display-name */
import React from 'react'
import { SiteCategory } from './SiteCategory'

interface CategoryParams {
  category: string
}

//React.memo化
const Category: React.VFC<CategoryParams> = React.memo(({ category }) => {
  return (
    <div className='my-1'>
      {category === 'ONEPIECE' && (
        <SiteCategory
          className='inline-block rounded-xl border  border-cyan-500 p-1 text-center  font-bold text-cyan-500 hover:bg-cyan-700 hover:text-white'
          text='ONE PIECE'
          href={`/post/categories/${category}`}
        />
      )}
      {category === '呪術廻戦' && (
        <SiteCategory
          className='inline-block rounded-xl border  border-purple-500 p-1 text-center  font-bold text-purple-500 hover:bg-purple-700 hover:text-white'
          text='呪術廻戦'
          href={`/post/categories/${category}`}
        />
      )}
      {category === '東京リベンジャーズ' && (
        <SiteCategory
          className='inline-block rounded-xl border  border-rose-500 p-1 text-center  font-bold text-rose-500 hover:bg-rose-700 hover:text-white'
          text='東京リベンジャーズ'
          href={`/post/categories/${category}`}
        />
      )}
      {category === 'キングダム' && (
        <SiteCategory
          className='inline-block rounded-xl border  border-yellow-500 p-1 text-center  font-bold text-yellow-500 hover:bg-yellow-700 hover:text-white'
          text='キングダム'
          href={`/post/categories/${category}`}
        />
      )}
    </div>
  )
})

export default Category
