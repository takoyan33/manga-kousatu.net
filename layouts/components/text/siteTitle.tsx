/* eslint-disable react/display-name */
import React from 'react'
import { SiteCategory } from './SiteCategory'

interface TitleParams {
  title: string
}

//React.memoåŒ–
export const siteTitle: React.VFC<TitleParams> = React.memo(({ title }) => {
  return <div className='my-1'>{title}</div>
})
