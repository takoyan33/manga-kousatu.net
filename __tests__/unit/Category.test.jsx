import { render, screen, fireEvent, userEvent } from '@testing-library/react'
import { Category } from '../../layouts/components/text/Category'
import '@testing-library/jest-dom'

describe('A1_Component > Category', () => {
  it('[A1_6_1] レンダリングされているか', () => {
    render(<Category category='ワンピース' />)
  })
  // it('[A1_6_2] buttonをクリックできるか', () => {
  //   render(<Category text='test' />)
  //   fireEvent.click(screen.getByRole('button', { name: 'open-button' }))
  // })

  //[A1_3_4] LINKをクリックできるか
  // const passwordChangeLink = screen.getByRole('link', { name: 'password-change' });
  // fireEvent.click(passwordChangeLink);
})
