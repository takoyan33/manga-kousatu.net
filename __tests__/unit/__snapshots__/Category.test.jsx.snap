// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A1_Component > Category [A1_6_2] Pageのスナップショット 1`] = `
<div>
  <div
    class="my-1"
  />
</div>
`;
